drools-compiler/src/main/java/org/drools/compiler/builder/impl/resources/ResourceHandler.java

```diff
@@ -38,21 +41,25 @@
 import java.util.Collection;
 import java.util.UUID;
 
-public abstract class Processor {
+public abstract class ResourceHandler {
     protected KnowledgeBuilderConfigurationImpl configuration;
     protected Collection<KnowledgeBuilderResult> results = new ArrayList<>();
     protected ReleaseId releaseId;
-    protected static final Logger logger = LoggerFactory.getLogger(Processor.class);
+    protected static final Logger logger = LoggerFactory.getLogger(ResourceHandler.class);
 
-    public Processor(KnowledgeBuilderConfigurationImpl configuration, ReleaseId releaseId) {
+    public ResourceHandler(KnowledgeBuilderConfigurationImpl configuration, ReleaseId releaseId) {
         this.configuration = configuration;
         this.releaseId = releaseId;
     }
 
-    public Processor(KnowledgeBuilderConfigurationImpl configuration) {
+    public ResourceHandler(KnowledgeBuilderConfigurationImpl configuration) {
         this.configuration = configuration;
     }
 
+    public abstract PackageDescr process(Resource resource, ResourceConfiguration configuration) throws DroolsParserException, IOException;
```

Comment:
Is the codepath where `ResourceConfiguration` is optional (`null`) common? I see multiple similar changes in the diff of this PR but if it is a possible value we can add an additional method like:
```suggestion
    public abstract PackageDescr process(Resource resource, ResourceConfiguration configuration) throws DroolsParserException, IOException;
 
    public PackageDescr process(Resource resource) throws DroolsParserException, IOException {
        return process(resource, null);
    }
```

Reply:
we can, I just have grown to dislike overloads of that kind since we have so many laying around in our public APIs. I'll do it anyway ;)